src/artemis/experiment_plans/tests/test_rotation_scan_plan.py

```diff
@@ -0,0 +1,169 @@
+        with patch.object(smargon.omega, "set", mock_omega_set):
+            RE(move_to_start_w_buffer(smargon, start_angle))
+
+    mock_velocity_set.assert_called_with(120)
+    mock_omega_set.assert_called_with(start_angle - OFFSET * DIRECTION)
+
+
+def __fake_read(obj, initial_positions, _):
+    initial_positions[obj] = 0
+    yield Msg("null", obj)
+
+
+def test_move_to_end(smargon: Smargon, RE):
+    scan_width = 153
+    mock_omega_set = MagicMock(return_value=Status(done=True, success=True))
+
+    with patch.object(smargon.omega, "set", mock_omega_set):
+        with patch(
+            "bluesky.preprocessors.__read_and_stash_a_motor",
+            __fake_read,
+        ):
+            # with patch.object(RE, "_read", MagicMock(return_value=0)):
+            RE(move_to_end_w_buffer(smargon, scan_width))
+
+    mock_omega_set.assert_called_with((scan_width + 0.1 + OFFSET) * DIRECTION)
+
+
+@patch("dodal.i03.active_device_is_same_type", lambda a, b: True)
+@patch("artemis.experiment_plans.rotation_scan_plan.rotation_scan_plan")
+def test_get_plan(
+    plan: MagicMock,
+    RE,
+    test_rotation_params,
+    smargon: Smargon,
+    zebra: Zebra,
+    eiger: EigerDetector,
+    detector_motion: DetectorMotion,
+    backlight: Backlight,
+):
+    eiger.stage = MagicMock()
+    eiger.unstage = MagicMock()
+    zebra.pc.armed.set(False)
+    with (
+        patch("dodal.i03.smargon", return_value=smargon),
+        patch("dodal.i03.eiger", return_value=eiger),
+        patch("dodal.i03.zebra", return_value=zebra),
+        patch("dodal.i03.backlight", return_value=backlight),
+        patch(
+            "artemis.experiment_plans.rotation_scan_plan.DetectorMotion",
+            return_value=detector_motion,
+        ),
+    ):
+        RE(get_plan(test_rotation_params, MagicMock()))
+
+    eiger.stage.assert_called()
+    eiger.unstage.assert_called()
+
+
+@patch("bluesky.plan_stubs.wait")
+def test_rotation_plan(
+    bps_wait: MagicMock,
+    RE,
+    test_rotation_params,
+    smargon: Smargon,
+    zebra: Zebra,
+    eiger: EigerDetector,
+    detector_motion: DetectorMotion,
+    backlight: Backlight,
+):
+    mock_omega_sets = MagicMock(return_value=Status(done=True, success=True))
+
+    mock_arm_disarm = MagicMock(
+        side_effect=zebra.pc.armed.set, return_value=Status(done=True, success=True)
+    )
+    zebra.pc.arm_demand.set = mock_arm_disarm
+
+    smargon.omega.velocity.set = mock_omega_sets
+    smargon.omega.set = mock_omega_sets
+
+    with patch(
+        "bluesky.preprocessors.__read_and_stash_a_motor",
+        __fake_read,
+    ):
+        RE(
+            rotation_scan_plan(
+                test_rotation_params, eiger, smargon, zebra, backlight, detector_motion
+            )
+        )
+
+    assert mock_omega_sets.call_count == 4
+
+
+@patch("artemis.experiment_plans.rotation_scan_plan.cleanup_plan")
+@patch("bluesky.plan_stubs.wait")
+def test_cleanup_happens(
```

Comment:
Nit: I would split this into two tests, one with the failure and one without. Probably also add a third that confirms it cleans up when no error happens